### DISCUSSION TITLE
Edit history

### DISCUSSION THREAD

**Discussion by ibnuh** [usememos/memos#798] [Ideas]
Hi! I'm a heavy user of google keep, but the lack of edit history always throws me off, and recently found out about memos and absolutely loving it!

Is there a plan for edit history? I would love to try and make a PR (I don't have golang experience)

**Comment by boojack** [comment]
The edit history sounds like a good idea. Please have a try.

**Reply by HubKing** [reply]
Memos really does need history feature. Memos currently has a "Save" button, whilst Google Keep does not have it and automatically saves the changes when the user navigates away. I guess this is because Memos must need an explicit save confirmation from the user because the change is irreversible. But it is annoying for a quick note app that the user has to press the "Save" button each time.

Most of the time, notes are just a small text, probably less than a few KB. Keeping previous copies instead of overwriting it, should not take that much space. Also, the history feature does not need to be an advanced one like other sites/apps that highlights changed parts. It can simply be complete copies of each save.

**Comment by almereyda** [comment]
This could even extend to being able to choose whether edits on public notes are public as well, or if only the last revision is shown publicly.
